lib/rubber/recipes/rubber/setup.rb

```diff
@@ -90,7 +91,9 @@ def enable_root_ssh(ip, initial_ssh_user)
     end
 
     task :_allow_root_ssh, :hosts => "#{initial_ssh_user}@#{ip}" do
-      rsudo "mkdir -p /root/.ssh && cp /home/#{initial_ssh_user}/.ssh/authorized_keys /root/.ssh/"
+      if initial_ssh_user != 'root'
+        rsudo "mkdir -p /root/.ssh && cp ~/.ssh/authorized_keys /root/.ssh/"
```

Comment:
~ is actually safe regardless of sudo status. ~ is the the home directory of the current logged in user. Its important to note that sudo does not changed the user who is logged in. as a result the safest thing to do here is to use ~. Assuming /home/{user} is wrong, because you can have home directories that are not the same as the username. $HOME is wrong for exactly the reason @tdg5 mentioned.
```
$ sudo -H echo ~
/home/james
```
$ sudo echo ~
/home/james
```
```